conformance/src/test/scala/za/co/absa/enceladus/conformance/interpreter/rules/CastingRuleSuite.scala

```diff
@@ -125,4 +127,60 @@ class CastingRuleSuite extends FunSuite with SparkTestBase with LoggerTestBase{
 
   }
 
+  test("Test Casting rule ok handling conversion from 'long' to 'string'") {
+    RuleValidators.validateTypeCompatibility(ruleName, columnName, LongType, "string")
+  }
+
+  test("Test Casting rule ok handling conversion from 'string' to 'long'") {
+    RuleValidators.validateTypeCompatibility(ruleName, columnName, StringType, "long")
+  }
+
+  test("Test Casting rule ok handling conversion from 'string' to 'date'") {
+    RuleValidators.validateTypeCompatibility(ruleName, columnName, StringType, "date")
+  }
+
+  test("Test Casting rule ok handling conversion from 'boolean' to 'decimal(10,2)'") {
+    RuleValidators.validateTypeCompatibility(ruleName, columnName, BooleanType, "decimal(10,2)")
+  }
+
+  test("Test Casting rule ok handling conversion from 'long' to 'timestamp'") {
+    RuleValidators.validateTypeCompatibility(ruleName, columnName, LongType, "timestamp")
+  }
+
+  test("Test Casting rule ok handling conversion from 'timestamp' to 'boolean'") {
+    RuleValidators.validateTypeCompatibility(ruleName, columnName, TimestampType, "boolean")
+  }
+
+  test("Test Casting rule ok handling conversion from 'date' to 'boolean'") {
+    RuleValidators.validateTypeCompatibility(ruleName, columnName, DateType, "boolean")
+  }
+
+  test("Test Casting rule failure handling conversion from 'long' to 'date'") {
+    val e = intercept[ValidationException] {
+      RuleValidators.validateTypeCompatibility(ruleName, columnName, LongType, "date")
+    }
+    assert(e.message.contains("conversion from 'long' to 'date' is not supported"))
+  }
+
+  test("Test Casting rule failure handling conversion from 'boolean' to 'date'") {
+    val e = intercept[ValidationException] {
+      RuleValidators.validateTypeCompatibility(ruleName, columnName, BooleanType, "date")
+    }
+    assert(e.message.contains("conversion from 'boolean' to 'date' is not supported"))
+  }
+
+  test("Test Casting rule failure handling conversion from 'decimal(10,2)' to 'date'") {
+    val e = intercept[ValidationException] {
+      RuleValidators.validateTypeCompatibility(ruleName, columnName, DecimalType(10, 2), "date")
+    }
+    assert(e.message.contains("conversion from 'decimal(10,2)' to 'date' is not supported"))
+  }
+
+  test("Test Casting rule failure handling conversion from 'boolean' to 'timestamp'") {
+    val e = intercept[ValidationException] {
+      RuleValidators.validateTypeCompatibility(ruleName, columnName, BooleanType, "timestamp")
+    }
+    assert(e.message.contains("conversion from 'boolean' to 'timestamp' is not supported"))
+  }
+
```

Comment:
Perhaps some `Double` to `Timestamp`, `Date` and `Long` and visa versa conversion tests.

Reply:
Ok, will add